### PR TITLE
Dont download on server

### PR DESCRIPTION
**Pull request by espindola**

### CODE REVIEW

**Comment by espindola** [comment]
This is a draft because it goes on top of #1430 and because we might decide that it requires a db migration. Currently we save the endpoint code in the database using the endpoint path. That is, given an endpoint `/dev/foo`, the endpoints table has an entry with `/dev/foo`.

Since endpoint paths are arbitrary, we cannot add other modules there without risking conflict. With this PR the path we use is `/dev/endpoints/foo.ts`. This will allow us to also store `/dev/models/bar.ts` and `/dev/lib/utils.ts` for example.

Without a migration, when moving to a chiseld with this patches, it would be necessary to run "chisel apply" again. Is that OK? If not I will code a migration and add it here.

**Comment by penberg** [comment]
@espindola I am fine with _not_ having a migration if it's just a matter of re-running `chisel apply`. @glommer what do you think

**Comment by glommer** [comment]
chisel apply again seems fine.

**Comment by espindola** [comment]
Cool, so this is ready for plain review then :-)

File: server/src/rpc.rs [hunk 1]
```diff
@@ -597,7 +596,8 @@ impl ChiselRpc for RpcService {
             }
             let mut endpoint_defs = vec![];
             let version_path_str = format!("/{}/", api_version);
-            for (path, _) in state.routes.iter() {
+            for (path, _) in state.sources.iter() {
+                // FIXME: filter out non endpoint sources
```
[the remark this answers is not in the document]
Comment (espindola) [reply server/src/rpc.rs:600]: Good catch. I will add a test, fix and rebase.

[the remark this answers is not in the document]
Comment (espindola) [reply server/src/rpc.rs]: done

**Comment by espindola** [comment]
> Other that one comment it looks fine to me, but I don't feel like I fully understand the source logic, isn't there someone else with a decent insight? For example I don't have sufficient knowledge to verify that this PR actually allows for

The reason for this PR is not running TSC on the server: https://github.com/chiselstrike/chiselstrike/pull/1431/files#diff-36a55fca956d2696a4fb89bc9847360169a01e70aa29641862a78ca833e9becfL56

**Pull request merged**